### PR TITLE
TEPHRA-112 Log basic stats when loading a transaction snapshot

### PR DESCRIPTION
**Pull request by ghelmling**
Log some basic size stats when transaction snapshot files are loaded.

### CODE REVIEW

File: tephra-core/src/main/java/co/cask/tephra/persist/HDFSTransactionStateStorage.java [hunk 1]
```diff
@@ -305,6 +308,79 @@ public boolean accept(Path path) {
   }
 
   /**
+   * Tracks the number of bytes read by the source InputStream.
+   */
+  private static class CountingInputStream extends InputStream {
```
Comment (chtyim) [inline tephra-core/src/main/java/co/cask/tephra/persist/HDFSTransactionStateStorage.java]: There is one from guava. Can we use that one?

Comment (ghelmling) [inline tephra-core/src/main/java/co/cask/tephra/persist/HDFSTransactionStateStorage.java]: Hmm, didn't realize.  Yes, I'll use that one.

**Comment by ghelmling** [comment]
Updated to use Guava CountingInputStream

**Comment by chtyim** [comment]
LGTM

**Pull request merged**